--- a/src/mesh/meshGFacePack.cpp
+++ b/src/mesh/meshGFacePack.cpp
@@ -8,6 +8,7 @@
 #include <queue>
 #include <map>
 #include <vector>
+#include <cstdint>
 #include "gmsh.h"
 #include "SPoint2.h"
 #include "SVector3.h"
--- a/contrib/QuadMeshingTools/qmtMeshGeometryOptimization.h
+++ b/contrib/QuadMeshingTools/qmtMeshGeometryOptimization.h
@@ -8,6 +8,7 @@
 #pragma once
 
 #include <float.h>
+#include <cstdint>
 #include "qmtMeshUtils.h"
 
 class SurfaceProjector;

